core/src/main/java/org/codice/imaging/nitf/core/NitfDataSource.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ */
+package org.codice.imaging.nitf.core;
+
+import java.util.List;
+import javax.imageio.stream.ImageInputStream;
+import org.codice.imaging.nitf.core.dataextension.NitfDataExtensionSegmentHeader;
+import org.codice.imaging.nitf.core.graphic.GraphicSegmentHeader;
+import org.codice.imaging.nitf.core.image.NitfImageSegmentHeader;
+import org.codice.imaging.nitf.core.label.LabelSegmentHeader;
+import org.codice.imaging.nitf.core.symbol.SymbolSegmentHeader;
+import org.codice.imaging.nitf.core.text.TextSegmentHeader;
+
+/**
+ * Data source for NITF file parts.
+ */
+public interface NitfDataSource {
+
+    /**
+     * Return the file-level header for the parsed file.
+     *
+     * @return the file-level header
+     */
+    NitfFileHeader getNitfHeader();
+
+    /**
+     * Return the image segment headers associated with this file.
+     *
+     * @return image segment headers
+     */
+    List<NitfImageSegmentHeader> getImageSegmentHeaders();
+
+    /**
+     * Return the list of image segment data.
+     *
+     * @return image segment data
+     */
+    List<ImageInputStream> getImageSegmentData();
```

Comment:
I think it would be preferable to have join the *SegmentHeader objects to their *SegmentData.  So you'd create an interface like:

interface NitfImageSegment {
  NitfImageSegmentHeader getHeader();
  ImageInputStream getData();
}

The current representation relies on the two lists being the same size, with the list index position being the key that joins the two parts of the image.  If the lists aren't in sync, I think the client will have a hard time matching headers to data and identify which image we don't have data for.